## About this document
This document contains the analysis and report for the Assessment of CASA0006-Data Science for Spatial Systems

Student ID: 21203086
Word count: XXX

--------------------------------------------------------------------------------------------------------------------------------

# TITLE XXXXXX

<a id="ID_intr"></a>
| [1.0 Introduction](#ID_intro) | [2.0 Literature review](#ID_lit) | [3.0 Research Question](#ID_RQ) | [4.0 Presentation of Data](#ID_data) | [5.0 Methodology](#ID_methods) | [6.0 Results](#ID_results) | [7.0 Discussion](#ID_disc) | [8.0 Conclusion](#ID_concl) | [Bibliography](#ID_biblio) | [Appendix](#ID_append) |

<a id="ID_intro"></a>
## 1.0 | Introduction 

This paper studies .... 
quick summary
text text

According to the Understanding Inequalities project ''the housing market has a key role in "sorting" poorer households into areas with the worst pollution, schools, crime and employ-ment. ''(Housing | Understanding Inequalities, 2022). This study will therefore investigate if there is a relationship between a variety of indicators for environmental, health and socioeco-nomic features that compose a census tract and house prices in the same area. The case study chosen is California and the baseline assumption is that pollution, unemployment, and poor health statistics will be reflected in the overall house price trend. 

<a id="ID_lit"></a>
## 2.0 | Literature Review

Text text text


<a id="ID_RQ"></a>
## 3.0 | Research Question

R.Q.: Can house prices be predicted from data on environmental, health, and socioeconomic information of a census tract?

    RQ1 | text 
    RQ2 | text?

<a id="ID_data"></a>
## 4.0 | Presentation of Data

The data used for this analysis were obtained from the following sources:
* [CalEnviroScreen 3.0 Results](https://data.ca.gov/dataset/calenviroscreen-3-0-results/resource/89b3f4e9-0bf8-4690-8c6f-715a717f3fae), downloaded 04 April 2021:

   - Who
   [FEDERAL HOUSING FINACE AGENCY](https://www.fhfa.gov/)
   - Where from
   - What
   - Data 
   - Explanation/description

* [House Price Index Census Tracts](https://www.fhfa.gov/DataTools/Downloads/Pages/House-Price-Index-Datasets.aspx), downloaded 04 April 2021:

   - Who
   - Where from
   - What
   - Data 
   - House price index definition

According to the data dictionary provided on the website:
"The FHFA House Price Index (HPI) is a broad measure of the movement of single-family house prices.  The HPI is a weighted, repeat-sales index, meaning that it measures average price changes in repeat sales or refinancings on the same properties. This information is obtained by reviewing repeat mortgage transactions on single-family properties whose mortgages have been purchased or securitized by Fannie Mae or Freddie Mac since January 1975."



For ease of access the data has been saved in this [GitHub repository](https://github.com/DaphneBdn/dsss_repo) 

<a id="ID_methods"></a>
## 5.0 | Methodology

Outline of methodology....text text text

5.1 Libraries

Loading the necessary libraries

In [3]:
# data manipulation and processing
#import os
import math
import numpy as np
import pandas as pd

# analysis

# Packages for visualisation
import matplotlib.cm as cm
import matplotlib.pyplot as plt
import seaborn as sns

# reproducability settings
random_state = 42 
np.random.seed(random_state)


5.1 Data Loading 

In [16]:
# Load in the House Price Index data 
#dfprice = pd.read_csv('https://github.com/DaphneBdn/dsss_repo/blob/main/data/raw/HPI_AT_BDL_tract.csv')
dfprice = pd.read_csv('data/raw/HPI_AT_BDL_tract.csv')
dfprice.head(5)

tract state_abbr  year  annual_change         hpi  hpi1990  hpi2000
0  1001020100         AL  1998            NaN  100.000000      NaN   100.82
1  1001020100         AL  1999          -5.60   94.400002      NaN    95.18
2  1001020100         AL  2000           5.06   99.180000      NaN   100.00
3  1001020100         AL  2001           7.08  106.210000      NaN   107.08
4  1001020100         AL  2002           7.11  113.760000      NaN   114.70

In [17]:
dfprice.tail(5)

tract state_abbr  year  annual_change     hpi  hpi1990  hpi2000
1048570  29007950100         MO  2003            NaN  100.00      NaN      NaN
1048571  29007950100         MO  2004          11.68  111.68      NaN      NaN
1048572  29007950100         MO  2005          -6.56  104.36      NaN      NaN
1048573  29007950100         MO  2006           7.93  112.63      NaN      NaN
1048574  29007950100         MO  2007          -9.61  101.81      NaN      NaN

In [14]:
#check the size of the dataframe
dfprice.shape

(1048575, 7)

In [15]:
# check the datatype of each column 
dfprice.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 7 columns):
 #   Column         Non-Null Count    Dtype  
---  ------         --------------    -----  
 0   tract          1048575 non-null  int64  
 1   state_abbr     1048575 non-null  object 
 2   year           1048575 non-null  int64  
 3   annual_change  968146 non-null   float64
 4   hpi            1017771 non-null  float64
 5   hpi1990        542092 non-null   float64
 6   hpi2000        900951 non-null   float64
dtypes: float64(4), int64(2), object(1)
memory usage: 56.0+ MB


After the first inspection of the data the key things to note are: 
* There are 1,048,575 entries in total
* There are multiple NA values: discuss approach, which columns
* years 1998-2007

#### subseting/filtering

* Extract census tracts of California only.
* Keep hpi 
* Year: Ideally keep only 2018, so as to match with the Enviro Dataset.
* If many NAs in 2018: perhaps combine the year before and after? 

In [26]:
dfCAall = dfprice[dfprice['state_abbr']=="CA"]

In [27]:
dfCAall.shape

(264546, 7)

In [38]:
columns = dfCAall.columns
columns
todrop=['annual_change', 'hpi1990','hpi2000']

In [39]:
dfCAall.drop(columns=todrop, inplace=True)
dfCAall

/opt/conda/envs/sds2021/lib/python3.9/site-packages/pandas/core/frame.py:4901: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


tract state_abbr  year        hpi
77893   6001400100         CA  1986  100.00000
77894   6001400100         CA  1987  103.27000
77895   6001400100         CA  1988  129.86000
77896   6001400100         CA  1989  142.83000
77897   6001400100         CA  1990  150.85001
...            ...        ...   ...        ...
342434  6115041102         CA  2017  175.52000
342435  6115041102         CA  2018  199.45000
342436  6115041102         CA  2019  211.89999
342437  6115041102         CA  2020  224.97000
342438  6115041102         CA  2021  260.67999

[264546 rows x 4 columns]

In [50]:
# check the number of tracts
dfCAall['tract'].nunique()

7720

In [40]:
dfCA2018= dfCAall[dfCAall['year']==2018]
dfCA2018

tract state_abbr  year         hpi
77925   6001400100         CA  2018   548.07001
77961   6001400200         CA  2018   647.09998
78004   6001400300         CA  2018  1491.56010
78047   6001400400         CA  2018  1559.68010
78083   6001400500         CA  2018   789.46997
...            ...        ...   ...         ...
342313  6115040901         CA  2018   196.80000
342345  6115041001         CA  2018   249.62000
342376  6115041002         CA  2018   250.35001
342406  6115041101         CA  2018   209.84000
342435  6115041102         CA  2018   199.45000

[7711 rows x 4 columns]

In [41]:
# change settings to display all rows
#pd.set_option('display.max_rows', None)

# investigate NA values by printing out the count of NA values per column 
dfCA2018.isna().sum().sort_values(ascending=False)

# change settings back 
#pd.reset_option('all')

hpi           863
tract           0
state_abbr      0
year            0
dtype: int64

In [42]:
dfCA2017= dfCAall[dfCAall['year']==2017]
dfCA2017.isna().sum().sort_values(ascending=False)

hpi           570
tract           0
state_abbr      0
year            0
dtype: int64

In [43]:
dfCA2019= dfCAall[dfCAall['year']==2019]
dfCA2019.isna().sum().sort_values(ascending=False)

hpi           551
tract           0
state_abbr      0
year            0
dtype: int64

In [35]:
dfCA3 = dfCAall[(dfCAall['year']==2017) | (dfCAall['year']==2018) |(dfCAall['year']==2019)]
dfCA3.isna().sum().sort_values(ascending=False)

hpi1990          8522
hpi2000          4226
annual_change    2484
hpi              1984
tract               0
state_abbr          0
year                0
dtype: int64

#### envir data

In [44]:
# Load in the Enviro data 
#dfenv = pd.read_csv('https://github.com/DaphneBdn/dsss_repo/blob/main/data/raw/calenviroscreen-3.0-results-june-2018-update.csv')
dfenv = pd.read_csv('data/raw/calenviroscreen-3.0-results-june-2018-update.csv')
dfenv.head(5)

Census Tract  Total Population California County    ZIP  \
0    6019001100              3174           Fresno   93706   
1    6071001600              6133    San Bernardino  91761   
2    6019000200              3167           Fresno   93706   
3    6077000801              6692       San Joaquin  95203   
4    6019001500              2206           Fresno   93725   

  Nearby City \n(to help approximate location only)   Longitude   Latitude  \
0                                            Fresno -119.781696  36.709695   
1                                           Ontario -117.618013  34.057780   
2                                            Fresno -119.805504  36.735491   
3                                          Stockton -121.314524  37.940517   
4                                            Fresno -119.717843  36.681600   

   CES 3.0 Score   CES 3.0 Percentile CES 3.0 \nPercentile Range  ...  \
0          94.09               100.00   95-100% (highest scores)  ...   
1          90.68                99.99   95-100% (highest scores)  ...   
2          85.97                99.97   95-100% (highest scores)  ...   
3          82.49                99.96   95-100% (highest scores)  ...   
4          82.03                99.95   95-100% (highest scores)  ...   

  Linguistic Isolation Pctl  Poverty  Poverty Pctl  Unemployment  \
0                     77.51     76.3         97.12          17.6   
1                     96.25     72.5         94.63          12.3   
2                     78.39     86.8         99.56          16.1   
3                     75.14     61.3         85.57          19.6   
4                     73.72     66.4         90.23          18.6   

   Unemployment Pctl  Housing Burden  Housing Burden Pctl  Pop. Char.   \
0              91.72            26.0                79.40        92.12   
1              71.82            34.1                93.75        87.44   
2              87.98            40.1                97.85        94.58   
3              94.97            21.1                63.54        86.70   
4              93.65            28.1                83.98        80.08   

   Pop. Char. Score  Pop. Char. Pctl  
0              9.55            99.70  
1              9.07            98.11  
2              9.81            99.99  
3              8.99            97.72  
4              8.30            92.76  

[5 rows x 57 columns]

In [45]:
# check size of dataframe
dfenv.shape

(8035, 57)

In [46]:
dfenv.isna().sum().sort_values(ascending=False)

Linguistic Isolation                                 242
Linguistic Isolation Pctl                            242
Low Birth Weight                                     222
Low Birth Weight Pctl                                222
Housing Burden Pctl                                  157
Housing Burden                                       157
Unemployment Pctl                                    155
Unemployment                                         155
CES 3.0 \nPercentile Range                           106
Pop. Char. Pctl                                      106
 CES 3.0 Percentile                                  106
CES 3.0 Score                                        106
Pop. Char. Score                                     106
Pop. Char.                                           106
Education                                             96
Education Pctl                                        96
Poverty Pctl                                          79
Poverty                        

In [49]:
# check the number of tracts in the combined df
dfenv['Census Tract'].nunique()

8035

NOTE 
According to [United Staes Census Bureau](https://www.census.gov/geographies/reference-files/2010/geo/state-local-geo-guides-2010/california.html), California has 8,057 census tracts

The Enviro dataset lists 8035
The HPI lists 7720 for all years, 7711 for 2018

In [54]:
# extract the column names
cols=dfenv.columns
print(cols)
print()
print(f"The Enviro 3.0 dataset has {len(cols)} columns")

Index(['Census Tract', 'Total Population', 'California County', 'ZIP',
       'Nearby City \n(to help approximate location only)', 'Longitude',
       'Latitude', 'CES 3.0 Score', ' CES 3.0 Percentile',
       'CES 3.0 \nPercentile Range', 'SB 535 Disadvantaged Community', 'Ozone',
       'Ozone Pctl', 'PM2.5', 'PM2.5 Pctl', 'Diesel PM', 'Diesel PM Pctl',
       'Drinking Water', 'Drinking Water Pctl', 'Pesticides',
       'Pesticides Pctl', 'Tox. Release', 'Tox. Release Pctl', 'Traffic',
       'Traffic Pctl', 'Cleanup Sites', 'Cleanup Sites Pctl',
       'Groundwater Threats', 'Groundwater Threats Pctl', 'Haz. Waste',
       'Haz. Waste Pctl', 'Imp. Water Bodies', 'Imp. Water Bodies Pctl',
       'Solid Waste', 'Solid Waste Pctl', 'Pollution Burden',
       'Pollution Burden Score', 'Pollution Burden Pctl', 'Asthma',
       'Asthma Pctl', 'Low Birth Weight', 'Low Birth Weight Pctl',
       'Cardiovascular Disease', 'Cardiovascular Disease Pctl', 'Education',
       'Education Pctl', 

<a id="ID_results"></a>
## 6.0 | Results

<a id="ID_disc"></a>
## 7.0 | Discussion 

<a id="ID_conc"></a>
## 8.0 | Conclusion

<a id="ID_biblio"></a>
## Bibliography

<a id="ID_append"></a>
## Appendix